cli/listener/src/main.rs

```diff
@@ -506,93 +506,109 @@ fn get_success_response(interval_prices: IntervalPrices) -> ResponseBox {
     .boxed()
 }
 
-/// Gets a response that will be sent to a requesting client for a price
-/// interval. Returns None if no method is passed.
+/// Gets a response that will be sent to a requesting client for an APY
+/// query.
 fn get_interval_price_request(
     db_connection: &Connection,
-    request_url: &str,
-) -> Option<ResponseBox> {
-    let parsed_url = match Url::parse(request_url) {
-        Ok(parsed_url) => parsed_url,
-        Err(_err) => {
-            return Some(get_error_response(ResponseError::new_bad_request(
-                "Error parsing Url",
-            )));
-        }
-    };
-    let method_name = parsed_url.path_segments()?.last()?;
-    if method_name != "apy" {
-        return Some(get_error_response(ResponseError::new_bad_request(
-            "Method not supported, use \"/apy?begin=<begin_date_iso8601>&end=<end_date_iso8601>\".",
-        )));
-    }
-    let parsed_request_url =
-        form_urlencoded::parse(parsed_url.query()?.as_bytes()).collect::<Vec<_>>();
-
-    let dates = match get_date_params(parsed_request_url) {
-        Ok(res) => Some(res),
-        Err(err) => return Some(get_error_response(err)),
-    }?;
-
+    range_date: &DateBeginEnd,
+) -> ResponseBox {
     let interval_prices = get_interval_price_for_period(
         db_connection
             .unchecked_transaction()
             .expect("Failed to create sqlite transaction."),
-        dates.start,
-        dates.end,
+        range_date.start,
+        range_date.end,
         SOLIDO_ID.to_owned(),
     );
     match interval_prices {
         // Error while getting the interval prices.
         Err(err) => {
             eprintln!("Internal Error when getting interval prices: {}", err);
-            Some(get_error_response(ResponseError::InternalServerError))
+            get_error_response(ResponseError::InternalServerError)
         }
         Ok(interval_prices_opt) => {
             if let Some(interval_prices) = interval_prices_opt {
                 // Got interval prices.
-                Some(get_success_response(interval_prices))
+                get_success_response(interval_prices)
             } else {
                 // No interval price could be calculated, probably because of few data points.
-                Some(get_error_response(ResponseError::new_bad_request(
+                get_error_response(ResponseError::new_bad_request(
```

Comment:
This is not really a bad request then, it’s not something that the client can do something about. Not sure what the best thing to return is here. Maybe 404. But let’s get this pull request merged first, we can tweak this later.